pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,384 @@
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	kafkamgmtclient "github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1/client"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+type options struct {
+	selectedClusterId string
+	clusterList       []v1.Cluster
+	//interactive                   bool
+	selectedCluster         v1.Cluster
+	clusterMachinePoolList  v1.MachinePoolList
+	existingMachinePoolList []v1.MachinePool
+	//validatedMachinePoolList      []v1.MachinePool
+	selectedClusterMachinePool    v1.MachinePool
+	requestedMachinePoolNodeCount int
+	accessKafkasViaPrivateNetwork bool
+	newMachinePool                v1.MachinePool
+	fleetshardParams              map[string]string
+
+	f *factory.Factory
+}
+
+const (
+	machinePoolId          = "kafka-standard"
+	machinePoolTaintKey    = "bf2.org/kafkaInstanceProfileType"
+	machinePoolTaintEffect = "NoExecute"
+	machinePoolTaintValue  = "standard"
+	//machinePoolInstanceType = "m5.2xlarge"
+	machinePoolInstanceType = "r5.xlarge"
+	machinePoolLabelKey     = "bf2.org/kafkaInstanceProfileType"
+	machinePoolLabelValue   = "standard"
+)
+
+func NewRegisterClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	//TODO add Localizer
+	cmd := &cobra.Command{
+		Use: "register-cluster",
+		// Short: f.Localizer.MustLocalize("registerCluster.cmd.shortDescription"),
+		Short:   "registerCluster.cmd.shortDescription",
+		Long:    "registerCluster.cmd.longDescription",
+		Example: "registerCluster.cmd.example",
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runRegisterClusterCmd(opts)
+		},
+	}
+
+	// TODO add Localizer and flags
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	// this flag will allow the user to pass the cluster id as a flag
+	opts.selectedClusterId = "123"
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", "cluster id")
+	//flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("registerCluster.flag.clusterId"))
+
+	return cmd
+}
+
+func runRegisterClusterCmd(opts *options) error {
+	getListClusters(opts)
+	runClusterSelectionInteractivePrompt(opts)
+	getMachinePoolList(opts)
+	selectAccessPrivateNetworkInteractivePrompt(opts)
+	registerClusterWithKasFleetManager(opts)
+
+	return nil
+}
+
+func getListClusters(opts *options) error {
+	// ocm client connection
+	conn, err := opts.f.Connection()
+	if err != nil {
+		return err
+	}
+	client, cc, err := conn.API().OCMClustermgmt()
+	if err != nil {
+		return err
+	}
+
+	resource := client.Clusters().List()
+	response, err := resource.Send()
+	if err != nil {
+		return err
+	}
+	clusters := response.Items()
+	var cls = []v1.Cluster{}
+	for _, cluster := range clusters.Slice() {
+		// TODO the cluster must be multiAZ
+		//if cluster.State() == "ready" && cluster.MultiAZ() == true {
+		if cluster.State() == "ready" {
+			cls = append(cls, *cluster)
+		}
+	}
+	opts.clusterList = cls
+	defer cc()
```

Comment:
This could leak the connection if an error occurs.
deferring closing the connection should be moved above: between line 88-89